### PR TITLE
add email field validation

### PR DESCRIPTION
**Pull request by Karlasa**
## Description
Adds frontend side email field validation

## Related Issue
#3094 Email field validation

### Verification Steps
email field on Account creation, account login, forgot password and checkout form has email specific validation and validates it in frontend side.

## Checklist
* I have added tests to cover my changes, if necessary.
* I have added translations for new strings, if necessary.
* I have updated the documentation accordingly, if necessary.

### CODE REVIEW

**Comment by fooman** [comment]
I suggest removing the regex in favour of a simple check for the presence of the @ symbol. A well laid out argument coming from here https://medium.com/hackernoon/the-100-correct-way-to-validate-email-addresses-7c4818f24643.

The regex seems to match what is currently in the back-end but the below does show that the back-end regex has needed more work over time where valid email addresses are still/were blocked.
https://github.com/magento/magento2/issues/12075
https://github.com/magento/magento2/pull/11576/files

In summary I would be in favour of keeping the frontend validation intent only and then if at a future time the back-end switches to email validation the frontend doesn't require any changes (and also wouldn't need to be kept in sync).

**Comment by sirugh** [comment]
See issue comment https://github.com/magento/pwa-studio/issues/3094#issuecomment-815879415
